core/src/main/scala/kafka/coordinator/group/GroupCoordinator.scala

```diff
@@ -1335,7 +1339,7 @@ private[group] class GroupCoordinator(
 
               // Failed to persist member.id of the given static member, revert the update of the static member in the group.
               group.updateMember(knownStaticMember, oldProtocols, oldRebalanceTimeoutMs, oldSessionTimeoutMs, null)
-              val oldMember = group.replaceStaticMember(groupInstanceId, newMemberId, oldMemberId)
+              val oldMember = group.replaceStaticMember(groupInstanceId, newMemberId, oldMemberId, clientId, clientHost)
```

Comment:
IIUC, `clientId` and `clientHost` are applicable for the replacing member. In this case, when the persistence of the new static member failed, we need to reset the client id and client host to the older one. WDYT?

Reply:
I think you are correct, I didn't realize it should reset these value. I will update it.